frontend/src/home/Home.tsx

```diff
@@ -124,79 +140,92 @@ export class Home extends React.Component<{}, AppState> {
-    const newFinishSemesterSeason = convertTermIdToSeason(
-      newFinishSemester.termId
-    );
-
-    let newState: AppState;
+      const finishClasses = Array.from(finishSemester.classes);
+      finishClasses.splice(destination.index, 0, movedClass);
+      const newFinishSemester: DNDScheduleTerm = {
+        ...finishSemester,
+        classes: finishClasses,
+      };
 
-    if (newStartSemesterYear === newFinishSemesterYear) {
-      // in same year
-      newState = {
-        ...this.state,
-        schedule: {
-          ...this.state.schedule,
-          yearMap: {
-            ...this.state.schedule.yearMap,
-            [newStartSemesterYear]: {
-              ...this.state.schedule.yearMap[newStartSemesterYear],
-              [newStartSemesterSeason]: newStartSemester,
-              [newFinishSemesterSeason]: newFinishSemester,
+      const newStartSemesterYear = convertTermIdToYear(newStartSemester.termId);
+      const newStartSemesterSeason = convertTermIdToSeason(
+        newStartSemester.termId
+      );
+      const newFinishSemesterYear = convertTermIdToYear(
+        newFinishSemester.termId
+      );
+      const newFinishSemesterSeason = convertTermIdToSeason(
+        newFinishSemester.termId
+      );
+
+      if (newStartSemesterYear === newFinishSemesterYear) {
+        // in same year
+        newState = {
+          ...this.state,
+          schedule: {
+            ...this.state.schedule,
+            yearMap: {
+              ...this.state.schedule.yearMap,
+              [newStartSemesterYear]: {
+                ...this.state.schedule.yearMap[newStartSemesterYear],
+                [newStartSemesterSeason]: newStartSemester,
+                [newFinishSemesterSeason]: newFinishSemester,
+              },
             },
           },
-        },
-      };
-    } else {
-      newState = {
-        ...this.state,
-        schedule: {
-          ...this.state.schedule,
-          yearMap: {
-            ...this.state.schedule.yearMap,
-            [newStartSemesterYear]: {
-              ...this.state.schedule.yearMap[newStartSemesterYear],
-              [newStartSemesterSeason]: newStartSemester,
-            },
-            [newFinishSemesterYear]: {
-              ...this.state.schedule.yearMap[newFinishSemesterYear],
-              [newFinishSemesterSeason]: newFinishSemester,
+        };
+      } else {
+        newState = {
+          ...this.state,
+          schedule: {
+            ...this.state.schedule,
+            yearMap: {
+              ...this.state.schedule.yearMap,
+              [newStartSemesterYear]: {
+                ...this.state.schedule.yearMap[newStartSemesterYear],
+                [newStartSemesterSeason]: newStartSemester,
+              },
+              [newFinishSemesterYear]: {
+                ...this.state.schedule.yearMap[newFinishSemesterYear],
+                [newFinishSemesterSeason]: newFinishSemester,
+              },
             },
```

Comment:
Are most of the above blobby changes due to husky/prettier reformatting? (I'm viewing from the `Files changed` tab for the PR).

Reply:
Yup! I moved this inside an if statement so the indenting got increased